src/restLink.ts

```diff
@@ -863,13 +863,12 @@ const resolver: Resolver = async (
   if (!isLeaf && isATypeCall) {
     // @type(name: ) is only supported inside apollo-link-rest at this time
     // so use the preAliasingNode as we're responsible for implementing aliasing!
-    // Also: exit early, since @type(name: ) && @rest() can't both exist on the same node.
     if (directives.rest) {
       throw new Error(
         'Invalid use of @type(name: ...) directive on a call that also has @rest(...)',
       );
     }
-    return addTypeToNode(preAliasingNode, directives.type.name);
```

Comment:
This looks like a significant flow change. Is it necessary to remove the early return here?

Reply:
To rephrase: why was the deleted comment line wrong? I thought that `@type(name:)` and `@rest(…)` shouldn't exist on the same node?

Reply:
You are right, the comment should stay and it was also possible to leave the flow intact by calling `copyExportVariables` before the return.